### PR TITLE
CDAP-14738 copy python transform from hydrator-plugins repository

### PR DESCRIPTION
**Pull request by aonischuk**
Copy the files which are related to python transform from hydrator-plugins/core-plugins.

Some classes were shared between JavaScriptTransform and PythonTransform. 
That's why I had to move them to hydrator-plugins/hydrator-commons. That change should as well be pushed to hydrator-plugins repo for this code to fully work.

### CODE REVIEW

File: widgets/PythonEvaluator-transform.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          "widget-attributes": {
+            "default": "def transform(record, emitter, context):\n    \"\"\"Transforms the provided input record into zero or more output records (as dictionaries) or errors (as an error dictionary).\n\n    Input records are available as a dictionary.\n\n    Args:\n        record (dict): The input record as a dictionary. e.g. to access a field called 'total' from the input record, use record['total'].\n        emitter (Emitter): an object that can be used to emit zero or more records (using the emitter.emit() method) or errors (using the emitter.emitError() method).\n        context (Context): an object that provides access to:\n                           1. CDAP Metrics - context.getMetrics().count('output', 1);\n                           2. CDAP Logs - context.getLogger().debug('Received a record');\n                           3. Lookups - context.getLookup('blacklist').lookup(input.id); or\n                           4. Runtime Arguments - context.getArguments.get('priceThreshold')\n\n    \"\"\"\n\n    emitter.emit(record)\n"
+          }
+        }
+      ]
+    }
+  ],
+  "emit-alerts": true,
+  "emit-errors": true,
+  "errorDataset": {
```
Comment (albertshau) [inline widgets/PythonEvaluator-transform.json]: remove this field, as error datasets are no longer supported (this shouldn't have been there in hydrator-plugins)

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,600 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright © 2015-2016 Cask Data, Inc.
```
Comment (albertshau) [inline pom.xml]: update the copyright to 2019

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-api</artifactId>
+      <version>${cdap.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-api-spark</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this should be needed

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,600 @@
...
+      <version>${cdap.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-etl-proto</artifactId>
+      <version>${cdap.version}</version>
+      <scope>provided</scope>
```
Comment (albertshau) [inline pom.xml]: this should only be used in tests and not in the source code, so the scope should be test

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,600 @@
...
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-etl-api</artifactId>
+      <version>${cdap.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-etl-api-spark</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,600 @@
...
+      <version>${cdap.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.cdap</groupId>
+      <artifactId>cdap-formats</artifactId>
```
Comment (albertshau) [inline pom.xml]: this shouldn't be needed

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,600 @@
...
+      <version>${guava.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>commons-lang</groupId>
+      <artifactId>commons-lang</artifactId>
+      <version>${commons-lang-version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-minicluster</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,600 @@
...
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${guava.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>commons-lang</groupId>
+      <artifactId>commons-lang</artifactId>
```
Comment (albertshau) [inline pom.xml]: Is this used in the python transform? I hope not.

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,600 @@
...
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>${junit.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockftpserver</groupId>
+      <artifactId>MockFtpServer</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,600 @@
...
+        </exclusion>
+        <exclusion>
+          <groupId>javax.servlet</groupId>
+          <artifactId>servlet-api</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>commons-httpclient</groupId>
+      <artifactId>commons-httpclient</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,600 @@
...
+      <exclusions>
+        <exclusion>
+          <groupId>commons-logging</groupId>
+          <artifactId>commons-logging</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.zookeeper</groupId>
+      <artifactId>zookeeper</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,600 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.python</groupId>
+      <artifactId>jython-standalone</artifactId>
+      <version>${jython.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>co.cask.hydrator</groupId>
+      <artifactId>format-common</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,600 @@
...
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.hydrator</groupId>
+      <artifactId>hydrator-common</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>commons-validator</groupId>
+      <artifactId>commons-validator</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: src/main/java/co/cask/hydrator/python/transform/PythonEvaluator.java [hunk 14]
```diff
@@ -0,0 +1,431 @@
+
+
+/*
+ * Copyright © 2015-2016 Cask Data, Inc.
```
Comment (albertshau) [inline src/main/java/co/cask/hydrator/python/transform/PythonEvaluator.java]: update year to 2019

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,600 @@
...
+        </exclusion>
+        <exclusion>
+          <groupId>com.sun.jersey</groupId>
+          <artifactId>jersey-core</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think any hadoop dependency is needed

**Comment by aonischuk** [comment]
@albertshau thanks for a detailed review. I have addressed all your comments in the follow up commit.
Also I removed every dependency from pom.xml which was possible to compile without.

Can you please re-review and if everything is okay to merge it?

**Pull request merged**